llvm/lib/Target/AArch64/AArch64MachineFunctionInfo.cpp

```diff
@@ -93,24 +79,16 @@ AArch64FunctionInfo::AArch64FunctionInfo(const Function &F,
   // TODO: skip functions that have no instrumented allocas for optimization
   IsMTETagged = F.hasFnAttribute(Attribute::SanitizeMemTag);
 
-  // BTI/PAuthLR may be set either on the function or the module. Set Bool from
-  // either the function attribute or module attribute, depending on what is
-  // set.
-  // Note: the module attributed is numeric (0 or 1) but the function attribute
-  // is stringy ("true" or "false").
-  auto TryFnThenModule = [&](StringRef AttrName, bool &Bool) {
-    if (F.hasFnAttribute(AttrName)) {
-      const StringRef V = F.getFnAttribute(AttrName).getValueAsString();
-      assert(V.equals_insensitive("true") || V.equals_insensitive("false"));
-      Bool = V.equals_insensitive("true");
-    } else if (const auto *ModVal = mdconst::extract_or_null<ConstantInt>(
-                   F.getParent()->getModuleFlag(AttrName))) {
-      Bool = ModVal->getZExtValue();
-    }
+  // BTI/PAuthLR are set on the function attribute.
+  auto TryFnAttr = [&](StringRef AttrName) -> bool {
+    if (F.hasFnAttribute(AttrName))
+      return F.getFnAttribute(AttrName).getValueAsBool();
+    else
+      return false;
   };
 
-  TryFnThenModule("branch-target-enforcement", BranchTargetEnforcement);
-  TryFnThenModule("branch-protection-pauth-lr", BranchProtectionPAuthLR);
+  BranchTargetEnforcement = TryFnAttr("branch-target-enforcement");
+  BranchProtectionPAuthLR = TryFnAttr("branch-protection-pauth-lr");
```

Comment:
What happens if you just call
```c++
F.getFnAttribute(AttrName).getValueAsBool();
```
without first checking `F.hasFnAttribute`? If you don't hit an assert, I think these can be as simple as:
```c++
BranchTargetEnforcement = F.getFnAttribute("branch-target-enforcement").getValueAsBool();
BranchProtectionPAuthLR = F.getFnAttribute("branch-protection-pauth-lr").getValueAsBool();
```
Maybe not, but can you give it a shot?

Reply:
you could also do `if (Attr = F.getFnAttribute(...)) Attr.getValueAsBool()` if not